feat: Add "Switching authentication flows" Dev Preview documentation (#5879)
aws-amplify/docs: src/pages/lib-v1/auth/switch-auth/q/platform/[platform].mdx

  description: `Learn more about how to switch between different auth flows in Amplify.`
};

import federation from '/src/fragments/lib-v1/v6/callout/auth-federation.mdx';

<Fragments fragments={{ js: federation }} />

For client side authentication there are three different flows:

1. `USER_SRP_AUTH`: The `USER_SRP_AUTH` flow uses the [SRP protocol (Secure Remote Password)](https://en.wikipedia.org/wiki/Secure_Remote_Password_protocol) where the password never leaves the client and is unknown to the server. This is the recommended flow and is used by default.

> 2. `USER_PASSWORD_AUTH`: The `USER_PASSWORD_AUTH` flow will send user credentials unencrypted to the backend. If you want to migrate users to Cognito using the "Migration" trigger and avoid forcing users to reset their passwords, you will need to use this authentication type because the Lambda function invoked by the trigger needs to verify the supplied credentials.

Reviewer: I will not say unencrypted because that is not correct, is encrypted by SSL protocol

Reviewer: ```suggestion
2. `USER_PASSWORD_AUTH`: The `USER_PASSWORD_AUTH` flow will send user credentials to the backend. This is needed in order to migrate users to a different Cognito User Pool.
```

> 3. `CUSTOM_AUTH`: The `CUSTOM_AUTH` flow is used to allow for a series of challenge and response cycles that can be customized to meet different requirements.

Reviewer 2: wondering if it make sense to callout that `CUSTOM_AUTH` is the underlying authFlow that cognito ends up taking but that the library will take in the response `CUSTOM_WITH_SRP` and `CUSTOM_WITHOUT_SRP` .
